boa/src/builtins/number/tests.rs

```diff
@@ -177,12 +178,255 @@ fn to_string() {
+    let zero_with_radix = forward(&mut engine, "zero_with_radix");
+    let num_smaller_than_radix = forward(&mut engine, "num_smaller_than_radix");
+    let num_with_radix = forward(&mut engine, "num_with_radix");
+    let fraction_with_radix = forward(&mut engine, "fraction_with_radix");
+
+    assert_eq!(zero_with_radix, String::from("0"));
+    assert_eq!(num_smaller_than_radix, String::from("9"));
+    assert_eq!(num_with_radix, String::from("5a"));
+    assert_eq!(fraction_with_radix, String::from("5a.e66666666668"));
+
+    assert_eq!(
+        String::from("NaN"),
+        forward(&mut engine, "Number(NaN).toString()")
+    );
+    assert_eq!(
+        String::from("Infinity"),
+        forward(&mut engine, "Number(1/0).toString()")
+    );
+    assert_eq!(
+        String::from("-Infinity"),
+        forward(&mut engine, "Number(-1/0).toString()")
+    );
+    assert_eq!(
+        String::from("0"),
+        forward(&mut engine, "Number(0).toString()")
+    );
+    assert_eq!(
+        String::from("9"),
+        forward(&mut engine, "Number(9).toString()")
+    );
+    assert_eq!(
+        String::from("90"),
+        forward(&mut engine, "Number(90).toString()")
+    );
+    assert_eq!(
+        String::from("90.12"),
+        forward(&mut engine, "Number(90.12).toString()")
+    );
+    assert_eq!(
+        String::from("0.1"),
+        forward(&mut engine, "Number(0.1).toString()")
+    );
+    assert_eq!(
+        String::from("0.01"),
+        forward(&mut engine, "Number(0.01).toString()")
+    );
+    assert_eq!(
+        String::from("0.0123"),
+        forward(&mut engine, "Number(0.0123).toString()")
+    );
+    // assert_eq!(String::from("111111111111111110000"), forward(&mut engine, "Number(111111111111111111111).toString()"));
+    // assert_eq!(String::from("1.1111111111111111e+21"), forward(&mut engine, "Number(1111111111111111111111).toString()"));
+    // assert_eq!(String::from("1.1111111111111111e+22"), forward(&mut engine, "Number(11111111111111111111111).toString()"));
+    assert_eq!(
+        String::from("0.00001"),
+        forward(&mut engine, "Number(0.00001).toString()")
+    );
+    assert_eq!(
+        String::from("0.000001"),
+        forward(&mut engine, "Number(0.000001).toString()")
+    );
+    // assert_eq!(String::from("1e-7"), forward(&mut engine, "Number(0.0000001).toString()"));
+    // assert_eq!(String::from("1.2e-7"), forward(&mut engine, "Number(0.00000012).toString()"));
+    // assert_eq!(String::from("1.23e-7"), forward(&mut engine, "Number(0.000000123).toString()"));
+    // assert_eq!(String::from("1e-8"), forward(&mut engine, "Number(0.00000001).toString()"));
+    // assert_eq!(String::from("1.2e-8"), forward(&mut engine, "Number(0.000000012).toString()"));
+    // assert_eq!(String::from("1.23e-8"), forward(&mut engine, "Number(0.0000000123).toString()"));
+
+    assert_eq!(
+        String::from("NaN"),
+        forward(&mut engine, "Number(NaN).toString(16)")
```

Comment:
I would change all these to have this syntax:
```suggestion
        "NaN",
        &forward(&mut engine, "Number(NaN).toString(16)")
```